File: solutions/javascript/hamming/1/hamming.js
export const compute = (firstDnaStrand, lastDnaStrand) => {
    if (firstDnaStrand.length !== lastDnaStrand.length) 
      throw new Error("strands must be of equal length");
    let hamming = 0;
  
    for (let i = 0; i < firstDnaStrand.length; i++) {
        if (firstDnaStrand[i] !== lastDnaStrand[i]) hamming++;
    }
    return hamming;
};

File: solutions/python/hamming/1/hamming.py
import re


def valid_nucleotide(nucleotide):
    return nucleotide in ['A', 'C', 'G', 'T', '']

def distance(strand_a, strand_b):
    if len(strand_a) != len(strand_b):
        raise ValueError("Strands must be of equal length.")

    if re.findall('[^ATCG]', strand_a) or re.findall('[^ATCG]', strand_b):
        raise TypeError("Invalid nucleotide.")

    return len([ nucleotide_a for nucleotide_a, nucleotide_b in zip(strand_a, strand_b)
             if nucleotide_a != nucleotide_b ])
